src/lib/code_imputation_functions.py

```diff
@@ -100,3 +100,48 @@ def impute_somalia_location_codes(user, data, location_configurations):
                                        SomaliaLocations.state_for_location_code(zone_str)),
                     Metadata.get_call_location()).to_dict()
             }, Metadata(user, Metadata.get_call_location(), time.time()))
+
+
+def impute_age_category(user, data, age_configurations):
+    # TODO: By accepting a list of age_configurations but then requiring that list to contain code schemes in a
+    #       certain order, it looks like we're providing more flexibility than we actually do. We should change this
+    #       to explicitly accept age and age_category configurations, which requires refactoring all of the
+    #       code imputation functions.
+    age_cc = age_configurations[0]
+    age_category_cc = age_configurations[1]
+    
+    age_categories = {
+        (10, 14): "10 to 14",
+        (15, 18): "15 to 18",
+        (19, 35): "19 to 35",
+        (36, 54): "36 to 54",
+        (55, 99): "55 to 99"
+    }
+
+    for td in data:
+        age_label = td[age_cc.coded_field]
+        age_code = age_cc.code_scheme.get_code_with_code_id(age_label["CodeID"])
```

Comment:
```suggestion
        age_code = age_cc.code_scheme.get_code_with_code_id(td[age_cc.coded_field]["CodeID"])
```

Reply:
Not sure I agree with this change - by pulling this out over two lines I was trying to show how we:
 - get the label, then
 - convert the label for a code
which is the mirror of the next step:
 - getting the categorised code 
 - converting to the categorised label

Reply:
Ahh sure makes sense. The steps make it easier to follow through. I am happy with either approach.

Reply:
will leave as-is unless @lukechurch suggests otherwise

Reply:
I prefer Alexander's original version